test/compress/issue-973.js

```diff
@@ -44,26 +54,43 @@ this_binding_collapse_vars: {
     input: {
         var c = a; c();
         var d = a.b; d();
+        var e = eval; e();
     }
     expect: {
         a();
         (0, a.b)();
+        (0, eval)();
     }
 }
 
-eval_direct_calls: {
+this_binding_side_effects: {
     options = {
-        side_effects: true,
-        collapse_vars: true
-    }
+        side_effects : true
+    };
     input: {
-        (0, eval)('');
-
-        var fn = eval;
-        fn('');
+        (function (foo) {
+            (0, foo)();
+            (0, foo.bar)();
+            (0, eval)('console.log(foo);');
+        }());
+        (function (foo) {
+            var eval = console;
+            (0, foo)();
+            (0, foo.bar)();
+            (0, eval)('console.log(foo);');
+        }());
     }
     expect: {
-        (0, eval)('');
-        (0, eval)('');
+        (function (foo) {
+            foo();
+            (0, foo.bar)();
+            (0, eval)('console.log(foo);');
+        }());
+        (function (foo) {
+            var eval = console;
+            foo();
+            (0, foo.bar)();
+            eval('console.log(foo);');
```

Comment:
I'm actually starting to worry now, because any `(0, a)()` can potentially be `eval()` which means I can't optimise it to `a()`, or am I just starting to see things and need :coffee: ?